### PR TITLE
chore: allow explicitly provided prevarbifiedlist and newarbifiedlist path arguments when arbifying

### PR DESCRIPTION
**Pull request by Jason-Wanxt**
No need to add `newArbifiedList` to update because it will just update the former file and no new file will generate.

### CODE REVIEW

File: src/lib/store.ts [hunk 1]
```diff
@@ -37,9 +37,11 @@ export const getPath = (l1ListName: string) => {
   return path;
 };
 
-export const getPrevList = (l1ListName: string): ArbTokenList | undefined => {
-  const path = getPath(l1ListName);
+export const getPrevList = (l1ListName: string, arbifiedList: string | null): ArbTokenList | undefined => {
+  // let prevArbTokenList;
```
Comment (chrstph-dvx) [inline src/lib/store.ts:41]: leftover comment

File: src/lib/store.ts [hunk 2]
```diff
@@ -37,9 +37,11 @@ export const getPath = (l1ListName: string) => {
...
+  const path = arbifiedList? arbifiedList: getPath(l1ListName);
```
Comment (chrstph-dvx) [inline src/lib/store.ts:42]: `const path = arbifiedList ?? getPath(l1ListName);`

File: src/commands/arbify.ts [hunk 3]
```diff
@@ -16,9 +16,10 @@ export const handler = async (argvs: Args) => {
   const { newList, l1ListName } = await arbifyL1List(argvs.tokenList, {
     includeOldDataFields,
     ignorePreviousList: argvs.ignorePreviousList,
+    prevArbifiedList: argvs.prevArbifiedList
   });
   let tokenList: ArbTokenList = newList;
-  const path = getPath(l1ListName);
+  const path = argvs.newArbifiedList? argvs.newArbifiedList: getPath(l1ListName);
```
Comment (chrstph-dvx) [inline src/commands/arbify.ts:22]: `const path = argvs.newArbifiedList ?? getPath(l1ListName);`

**Review by chrstph-dvx — COMMENTED** [review]
LGTM, just some small comments.

Also if you could run prettier please

File: src/lib/token_list_gen.ts [hunk 4]
```diff
@@ -411,18 +413,21 @@ export const updateArbifiedList = async (
...
 ) => {
   const arbTokenList = await getTokenListObj(pathOrUrl);
   removeInvalidTokensFromList(arbTokenList);
-  const path =
-    process.env.PWD +
-    '/src/ArbTokenLists/' +
-    listNameToFileName(arbTokenList.name);
+  const path = prevArbifiedList
+    ? prevArbifiedList
+    : process.env.PWD +
```
Comment (chrstph-dvx) [inline src/lib/token_list_gen.ts:428]: Sorry, I overlooked it on the first pass. Can we use `TOKENLIST_DIR_PATH` instead?
```typescript
const path = prevArbifiedList ?? TOKENLIST_DIR_PATH + listNameToFileName(arbTokenList.name);
```

**Review by chrstph-dvx — COMMENTED** [review]
Just one small fix to go

**Review by chrstph-dvx — APPROVED** [review]
Good job!

**Review by chrstph-dvx — APPROVED** [review]

**Review by DZGoldman — CHANGES_REQUESTED** [review]
awesome! let's get https://github.com/OffchainLabs/arb-token-lists/pull/69 merged + on this branch before we merge this one, so we can see it tested.

Also can we update the `update_all` tasks to use these new args / see it in action?

File: src/commands/update.ts [hunk 5]
```diff
@@ -21,6 +21,7 @@ export const handler = async (argvs: Args) => {
   let tokenList: ArbTokenList = newList;
 
   if (argvs.includePermitTags) tokenList = await addPermitTags(tokenList);
+  console.log(path);
```
Comment (dewanshparashar) [inline src/commands/update.ts]: Are we sure we're keeping this log?

**Comment by Jason-Wanxt** [comment]
This should be removed in this [commit](
https://github.com/OffchainLabs/arb-token-lists/pull/75/commits/37eae187ea5d484ebae2875767e320637b3f5a33
)

Dewansh ***@***.***>于2023年2月6日 周一19:57写道：

File: update_all [hunk 6]
```diff
@@ -9,19 +9,20 @@ fi
 
 # arb1
 yarn fullList --l2NetworkID 42161
-yarn arbify --l2NetworkID 42161 --tokenList https://gateway.ipfs.io/ipns/tokens.uniswap.org
-yarn arbify --l2NetworkID 42161 --tokenList https://www.gemini.com/uniswap/manifest.json
-yarn arbify --l2NetworkID 42161 --tokenList https://api.coinmarketcap.com/data-api/v3/uniswap/all.json
+yarn arbify --l2NetworkID 42161 --tokenList https://gateway.ipfs.io/ipns/tokens.uniswap.org --newArbifiedList ./src/ArbTokenLists/arbed_uniswap_labs.json
```
Comment (chrstph-dvx) [inline update_all:12]: Can we update the CI too, so we generate them with the proper name?

**Review by chrstph-dvx — CHANGES_REQUESTED** [review]
LGTM, just one more change

File: update_all (hunk 6 above)
Comment (Jason-Wanxt) [reply update_all:12]: Sure

**Review by chrstph-dvx — APPROVED** [review]

**Review by dewanshparashar — APPROVED** [review]

**Review by DZGoldman — APPROVED** [review]
looks good! 

Part of me thinks we should just get rid of the `getPath = (l1ListName: string)` method entirely, as it's kind of inherently ill-advised and cursed. IOW: --newArbifiedList would just always be required, and --prevArbifiedList would be optional / just produce a list with a version number of 1 if null. (Maybe an explicit --prevArbifiedList="NO_PREV" should be required so this doesn't become footgun).

Approving b/c it's mergable as is, if you prefer to merge and further the discussion/refactor in a new PR 👍

**Comment by Jason-Wanxt** [comment]
Sure, merged and will solve this in the next pr.

**Pull request merged**